torch/distributions/categorical.py

```diff
@@ -68,6 +68,14 @@ def probs(self):
     def param_shape(self):
         return self._param.size()
 
+    @property
+    def mean(self):
+        return torch.Tensor([float('nan')]).expand(self._extended_shape())
```

Comment:
To correctly set precision, `.is_cuda`, and `.cuda_device` you should use `.new()`:
```py
def mean(self):
    return self._param.new([float('nan')]).expand(self._extended_shape())
```